From 21881a71eaffc2a7b35deb7d2d01c2dfb622649e Mon Sep 17 00:00:00 2001
From: Alex <leha-bot@users.noreply.github.com>
Date: Sun, 26 Apr 2020 14:37:54 +0300
Subject: [PATCH 3/4] CMakeLists: add VERSION to project() and package

So the library consumers via find_package(Microsoft.GSL 3.0.2) could determine the actual library version.
---
 CMakeLists.txt | 14 +++++++++++++-
 1 file changed, 13 insertions(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 09f6777..0a070bf 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1,6 +1,6 @@
 cmake_minimum_required(VERSION 3.1.3)
 
-project(GSL CXX)
+project(GSL VERSION 3.0.2 LANGUAGES CXX)
 
 include(ExternalProject)
 find_package(Git)
@@ -100,6 +100,18 @@ install(
 install(EXPORT GSLConfig DESTINATION share/GSL/cmake)
 export(TARGETS GSL FILE GSLConfig.cmake)
 
+# Add find_package() versioning support. The version for
+# generated Microsoft.GSLConfigVersion.cmake will be used from
+# last project() command. The version's compatibility is set between all
+# minor versions (as it was in prev. GSL releases).
+include(CMakePackageConfigHelpers)
+write_basic_package_version_file(
+    ${CMAKE_CURRENT_BINARY_DIR}/Microsoft.GSLConfigVersion.cmake
+    COMPATIBILITY SameMajorVersion
+    ARCH_INDEPENDENT
+)
+install(FILES ${CMAKE_CURRENT_BINARY_DIR}/Microsoft.GSLConfigVersion.cmake DESTINATION ${CMAKE_INSTALL_DATADIR}/cmake/Microsoft.GSL)
+
 # Add Microsoft.GSL::GSL alias for GSL so that dependents can be agnostic about
 # whether GSL was added via `add_subdirectory` or `find_package`
 add_library(Microsoft.GSL::GSL ALIAS GSL)
-- 
2.30.2

